src/mobile/MobileToolbarBack.js

```diff
@@ -6,21 +6,24 @@ export default class MobileToolbarBack extends Component {
 
 	static propTypes = {
 		backBtnBarTitle: PropTypes.string,
+		actions: PropTypes.object,
+		to: PropTypes.string,
 	};
-
+	goBack(e) {
+		const { actions } = this.props;
+		e.preventDefault();
+		actions.resetAssetPickerFilter();
```

Comment:
@nuruddeensalihu I guess boris meant it should not be called here `Inside goBack`, but somewhere else, maybe in componentWillMount of relevant Component